client/scripts/models/NotificationSubscription.ts

```diff
@@ -53,11 +53,11 @@ class NotificationSubscription {
       json.is_active,
       json.created_at,
       json.immediate_email,
-      json.Chain,
-      json.ChainEventType,
-      json.OffchainComment,
-      json.OffchainCommunity,
-      json.OffchainThread,
+      json.Chain || json.chain_id,
```

Comment:
I am a bit wary of these model type changes, because json.Chain should theoretically be a `Chain` model and not just an id. But it seems not to break anything, so we should at least document this by using proper types on these model objects.

Reply:
I just checked and none of the places where NotificationSubscription.fromJSON() is used actually provides json.Chain on **master**:
For `createSubscription`:
```
{
  category_id: "new-thread-creation"
  chain_entity_id: null
  chain_event_type_id: null
  chain_id: "ethereum"
  created_at: "2022-02-14T21:18:08.545Z"
  id: 289495
  immediate_email: false
  is_active: true
  object_id: "ethereum"
  offchain_comment_id: null
  offchain_thread_id: null
  subscriber_id: 12113
  updated_at: "2022-02-14T21:18:08.545Z"
}
```
and for `viewNotifications`:

```
{
  ChainEventType: null
  NotificationsReads: []
  category_id: "new-reaction"
  chain_entity_id: null
  chain_event_type_id: null
  chain_id: "ethereum"
  created_at: "2021-12-14T21:04:33.903Z"
  id: 229531
  immediate_email: false
  is_active: true
  object_id: "comment-12631"
  offchain_comment_id: 12631
  offchain_thread_id: null
  subscriber_id: 12113
  updated_at: "2021-12-14T21:04:33.903Z"
}
```

As you can see only `chain_id` is present in all payloads. Also, for some reason, we use snake_case for backend models and camelCase for front-end models which is also why there are several logical or operations to check for the different types. I think a general cleanup and synchronization of front-end + back-end models is needed. Though that is probably out of the scope of this PR.

Reply:
Alright, so I ended up removing the Chain from `viewSubscriptions` since every usage seemed to use `NotificationSubscription.Chain.id`. None of the usages of NotificationsSubscription actually needed the full Chain record/instance just the id. Notifications settings are actually broken in production right now because of the previous notifications refactor. The changes I made for this fix resolve that bug so another reason for this PR to go out ASAP. Please QA notifications settings if you have the time. Works on my end but I can't change a lot of notification settings...

Reply:
Will do.